src/database_connection.cpp

```diff
@@ -318,6 +378,27 @@ QueryResult DatabaseConnection::query(MultiFilterSelect query) {
   return result;
 }
 
+QueryResult DatabaseConnection::query(MultiVariantFilterSelect query) {
+  pqxx::result pqxx_result = performSelect(query);
+  QueryResult result{};
+  result.table = query.table;
+  auto count = query.fields.size();
+
+  for (auto row : pqxx_result) {
+    int index = 0;
+    for (const auto &field : row) {
```

Comment:
in c++, is there for (int i = 0; i < object.length(); i++) { // function }
It looks weird to have index outside of for loop and it's used inside like that.

Reply:
There is.. I was half way convinced to use a plain for loop for that reason.